core-bundle/src/Controller/ContentElement/DownloadsController.php

```diff
@@ -222,14 +222,22 @@ private function getPreviews(FilesystemItem $filesystemItem, ContentModel $model
         }
     }
 
-    private function handleDownload(Request $request): void
+    private function handleDownload(Request $request, ContentModel $model): void
     {
         $response = $this->fileDownloadHelper->handle(
             $request,
             $this->filesStorage,
-            function (FilesystemItem $item, array $context): Response|null {
+            function (FilesystemItem $item, array $context) use ($model): Response|null {
+                $id = $context['id'] ?? null;
+
+                // Do not handle downloads from other DownloadController
+                // elements on the same page (see #5568)
+                if ($id !== $model->id) {
+                    return new Response('', Response::HTTP_NO_CONTENT);
+                }
+
                 if (
-                    null === ($model = $this->getContaoAdapter(ContentModel::class)->findById($context['id'] ?? null)) ||
+                    null === ($model = $this->getContaoAdapter(ContentModel::class)->findById($id)) ||
                     !$this->getFilesystemItems($model)->any(static fn (FilesystemItem $listItem) => $listItem->getPath() === $item->getPath())
```

Comment:
How is this better than directly passing the `$filesystemItems` as done in #5571?

Reply:
NVM, I have just read your https://github.com/contao/contao/pull/5571#issuecomment-1348183251. 👍